### PR TITLE
Add documentation website

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, lsif-java didn't have any documentation for end-users. We
only had documentation for contributors. This commit adds a website that
includes user-facing documentation on how to run lsif-java with
automatic indexing and manual indexing.

### CODE REVIEW

**Comment by olafurpg** [comment]
Live preview at https://sourcegraph.github.io/lsif-java

Just opening this PR to close my day and collect early feedback. I will review the change again tomorrow and likely clean up a few small things.

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+Please open an issue if your build tool is not listed in the table below. Feel
+free to subscribe to the tracking issues to receive updates on your build tool.
+
+| Build tool | Support | Tracking issue                                                                   |
+| ---------- | ------- | -------------------------------------------------------------------------------- |
+| Gradle     | ✅      |                                                                                  |
+| Maven      | ✅      |                                                                                  |
+| Bazel      | ❌      | [sourcegraph/lsif-java#88](https://github.com/sourcegraph/lsif-java/issues/88)   |
+| Buck       | ❌      | [sourcegraph/lsif-java#99](https://github.com/sourcegraph/lsif-java/issues/99)   |
+| sbt        | ❌      | [sourcegraph/lsif-java#110](https://github.com/sourcegraph/lsif-java/issues/110) |
```
Comment (Strum355) [inline docs/getting-started.md:133]: image

File: docs/manual-configuration.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+fails.
+
+## Overview
+
+Indexing a codebase consists of two independent steps:
+
+- Compile the codebase with the SemanticDB compiler plugin.
+- Generate LSIF index from SemanticDB files.
+
+![A three stage pipeline that starts with a list of Java sources, creates a list of SemanticDB files that then become a single LSIF index.](docs/img/semanticdb-javac-pipeline.svg)
```
Comment (Strum355) [inline docs/manual-configuration.md:18]: Not sure if its just the live demo of the website, but this image doesn't render currently

File: docs/manual-configuration.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+```scala
+Compile / javacOptions += {
+    val sourceroot = (ThisBuild / baseDirectory).value
+    val targetroot = (ThisBuild / target).value / "semanticdb-targetroot"
+    s"-Xplugin:semanticdb -sourceroot:$sourceroot -targeroot:$sourceroot"
+}
+```
+
+## Step 4: compile the codebase
```
Comment (Strum355) [inline docs/manual-configuration.md:105]: Step 3 disappeared :ghost:

File: docs/manual-configuration.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+Compile all source files in the codebase once the compiler setting has been
+configured. The exact command depends on your build tool. Below are some
+examples:
+
+- Gradle: `./gradlew clean compileTestJava`
```
Comment (Strum355) [inline docs/manual-configuration.md:111]: I know `compileTestJava` technically invokes `compileJava` as a dependency, so unless explicitly invoking both would result in indexing twice, I'd personally also include `compileJava` so users mightnt be left wondering why we're only suggesting to run `compileTestJava` as an example :smile: 

```suggestion
- Gradle: `./gradlew clean compileJava compileTestJava`
```

File: docs/manual-configuration.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+## Overview
+
+Indexing a codebase consists of two independent steps:
+
+- Compile the codebase with the SemanticDB compiler plugin.
+- Generate LSIF index from SemanticDB files.
+
+![A three stage pipeline that starts with a list of Java sources, creates a list of SemanticDB files that then become a single LSIF index.](docs/img/semanticdb-javac-pipeline.svg)
+
+## Step 1: Add SemanticDB compiler plugin to the classpath
```
Comment (Strum355) [inline docs/manual-configuration.md:20]: Should we have consistent first letter capitalization or is this intentionally capitalized?

**Review by Strum355 — APPROVED** [review]
Really like having this website, great idea :tada: 
Some minor but not-really-blocking comments :slightly_smiling_face:

**Review by olafurpg — COMMENTED** [review]
Thank you for the review!

**Pull request merged**